### PR TITLE
doc: add a figure, and description updates for the mesh model callbacks

### PR DESCRIPTION
**Pull request by Avalei**
Created a figure to illustrate the mesh model callbacks. Added minor text improvements to the mesh model callbacks section. Updated the vendor model development overview title to include "vendor" in the title.

Reference: [NCSDK-9754](https://projecttools.nordicsemi.no/jira/browse/NCSDK-9754)

Signed-off-by: Deidre Casey <deidre.casey@nordicsemi.no>

### CODE REVIEW

**Review by mia-ko — APPROVED** [review]

File: doc/nrf/ug_bt_mesh_vendor_model_dev_overview.rst [hunk 1]
```diff
@@ -289,6 +289,11 @@ Defining model callbacks
 The Access API provides a set of callbacks that are called when certain events occur.
```
Comment (omkar3141) [inline doc/nrf/ug_bt_mesh_vendor_model_dev_overview.rst:289]: The name of the file says vendor model development overview, and the actual page title says "Model development overview". I think it is better to be renamed as - "Vendor model development overview".

File: doc/nrf/images/bt_mesh_model_callbacks.svg [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+			<rect x="0" y="513.854" width="126.309" height="34.0157" class="st16"/>
+			<text x="6.54" y="526.66" class="st12">The model restores its state <tspan x="6.18" dy="1.38em" class="st15">from the persistent storage.</tspan></text>		</g>
+		<g id="shape1109-160" transform="translate(379.85,-188.536)">
+			<title>Nordic Lake.1109</title>
+			<desc>All Access APIs are ready for use. The model may start its be...</desc>
+			<rect x="0" y="513.854" width="145.991" height="34.0157" class="st16"/>
+			<text x="6.43" y="526.66" class="st12">All Access APIs are ready for use. <tspan x="4.84" dy="1.38em" class="st15">The model may start its behavior.</tspan></text>		</g>
+		<g id="shape1110-164" transform="translate(399.533,-63.6088)">
+			<title>Nordic Lake.1110</title>
+			<desc>The model resets its state from the persistent storage.</desc>
```
Comment (omkar3141) [inline doc/nrf/images/bt_mesh_model_callbacks.svg:255]: Suggestion: "Model resets its state and clears stored data (if any) from persistent storage"

**Review by omkar3141 — CHANGES_REQUESTED** [review]

**Review by Balaklaka — APPROVED** [review]

File: doc/nrf/ug_bt_mesh_vendor_model_dev_overview.rst (hunk 1 above)
Comment (Avalei) [reply doc/nrf/ug_bt_mesh_vendor_model_dev_overview.rst:289]: It looks like there are a few other instances in the documentation where we refer to it just as "model", such as "Creating a new model", "Defining a model identifier", and "Adding the model to the node composition data". 

Does it make sense to change all these instances to "vendor model"? Does it sound misleading if we do not include "vendor"?

Comment (omkar3141) [reply doc/nrf/ug_bt_mesh_vendor_model_dev_overview.rst:289]: I feel the title update is sufficient.  At other places, the context clarifies the meaning.

Comment (Avalei) [reply doc/nrf/ug_bt_mesh_vendor_model_dev_overview.rst:289]: Title is now updated to include "vendor".

File: doc/nrf/images/bt_mesh_model_callbacks.svg (hunk 2 above)
Comment (omkar3141) [reply doc/nrf/images/bt_mesh_model_callbacks.svg:255]: It needs to be erased manually, in the sense, an implementer should write corresponding code as shown on the last arrow of the diagram.

**Review by omkar3141 — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Pull request merged**